From 6b402fb9e519d5d76da339a2ac198c0cc1d2b6d7 Mon Sep 17 00:00:00 2001
From: Steve Longerbeam <steve_longerbeam@mentor.com>
Date: Mon, 16 Mar 2015 16:37:09 -0700
Subject: [PATCH 635/668] media: imx6 camera: adv7180: Enable NEWAVMODE

Modify FSL's register set to enable NEWAVMODE in register 0x31. With
that we don't want to touch registers 0x31-0x32 and 0xE5-0xEA, since
those are for manually controlling VS position when NEWAVMODE is disabled.

Based on linux-meibp 0673a7ccc66f75927d922c23d91748eafe867727.

Signed-off-by: Steve Longerbeam <steve_longerbeam@mentor.com>
---
 drivers/staging/media/imx6/capture/adv7180.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/media/imx6/capture/adv7180.c b/drivers/staging/media/imx6/capture/adv7180.c
index 3f00ebd..fa788f0 100644
--- a/drivers/staging/media/imx6/capture/adv7180.c
+++ b/drivers/staging/media/imx6/capture/adv7180.c
@@ -305,7 +305,7 @@ static const struct adv7180_reg_tbl_t adv7180_fsl_init_reg[] = {
 	{ 0x25, 0x00 }, { 0x26, 0x00 }, { 0x27, 0x58 }, { 0x28, 0x00 },
 	{ 0x29, 0x00 }, { 0x2A, 0x00 }, { 0x2B, 0xE1 }, { 0x2C, 0xAE },
 	{ 0x2D, 0xF4 }, { 0x2E, 0x00 }, { 0x2F, 0xF0 }, { 0x30, 0x00 },
-	{ 0x31, 0x12 }, { 0x32, 0x41 }, { 0x33, 0x84 }, { 0x34, 0x00 },
+	{ 0x31, 0x02 }, { 0x34, 0x00 },
 	{ 0x35, 0x02 }, { 0x36, 0x00 }, { 0x37, 0x01 }, { 0x38, 0x80 },
 	{ 0x39, 0xC0 }, { 0x3A, 0x10 }, { 0x3B, 0x05 }, { 0x3C, 0x58 },
 	{ 0x3D, 0xB2 }, { 0x3E, 0x64 }, { 0x3F, 0xE4 }, { 0x40, 0x90 },
@@ -348,8 +348,7 @@ static const struct adv7180_reg_tbl_t adv7180_fsl_init_reg[] = {
 	{ 0xD9, 0x08 }, { 0xDA, 0x3C }, { 0xDB, 0x9B }, { 0xDC, 0xAC },
 	{ 0xDD, 0x4C }, { 0xDE, 0x00 }, { 0xDF, 0x00 }, { 0xE0, 0x14 },
 	{ 0xE1, 0x80 }, { 0xE2, 0x80 }, { 0xE3, 0x80 }, { 0xE4, 0x80 },
-	{ 0xE5, 0x25 }, { 0xE6, 0x44 }, { 0xE7, 0x63 }, { 0xE8, 0x65 },
-	{ 0xE9, 0x14 }, { 0xEA, 0x63 }, { 0xEB, 0x55 }, { 0xEC, 0x55 },
+	{ 0xEB, 0x55 }, { 0xEC, 0x55 },
 	{ 0xEE, 0x00 }, { 0xEF, 0x4A }, { 0xF0, 0x44 }, { 0xF1, 0x0C },
 	{ 0xF2, 0x32 }, { 0xF3, 0x00 }, { 0xF4, 0x3F }, { 0xF5, 0xE0 },
 	{ 0xF6, 0x69 }, { 0xF7, 0x10 }, { 0xF8, 0x00 }, { 0xF9, 0x03 },
-- 
2.4.10

